x-pack/test/functional/services/ml/job_table.ts

```diff
@@ -50,6 +50,28 @@ export function MachineLearningJobTableProvider(
   const retry = getService('retry');
 
   return new (class MlJobTable {
+    public async clickJobRowCalendar(jobId: string, calendarId: string) {
+      await this.withDetailsOpen(jobId, async function clickJobRowCalendar() {
+        const calendarSelector = `${jobId}-${calendarId}`;
+        await testSubjects.existOrFail(calendarSelector, {
+          timeout: 3_000,
+        });
+        await testSubjects.click(calendarSelector, 3_000);
+      });
+    }
+
+    public async assertJobRowCalendars(jobId: string, expectedCalendars: string[]) {
+      for await (const expectedCalendar of expectedCalendars)
+        await this.withDetailsOpen(jobId, async function verifyJobRowCalendar() {
+          const calendarSelector = `${jobId}-${expectedCalendar}`;
+          await testSubjects.existOrFail(calendarSelector, {
+            timeout: 3_000,
+          });
+          const calendarVisibleText = await testSubjects.getVisibleText(calendarSelector);
+          expect(calendarVisibleText).to.be(expectedCalendar);
+        });
+    }
```

Comment:
This could be slightly optimized: instead of opening and closing the details of the same job row for every calendar to check, the job row could be expanded once and then all calendars are checked. So basically changing the order of the `withDetailsOpen` and the `for await`.